src/backend/oneapi/svd.cpp

```diff
@@ -38,15 +38,31 @@ void svdInPlace(Array<Tr> &s, Array<T> &u, Array<T> &vt, Array<T> &in) {
     int64_t LDU   = uStrides[1];
     int64_t LDVt  = vStrides[1];
 
-    int64_t scratch_size = ::oneapi::mkl::lapack::gesvd_scratchpad_size<T>(
-        getQueue(), ::oneapi::mkl::jobsvd::vectors,
-        ::oneapi::mkl::jobsvd::vectors, M, N, LDA, LDU, LDVt);
-    Array<T> scratchpad = createEmptyArray<T>(af::dim4(scratch_size));
-
-    ::oneapi::mkl::lapack::gesvd(
-        getQueue(), ::oneapi::mkl::jobsvd::vectors,
-        ::oneapi::mkl::jobsvd::vectors, M, N, *in.get(), LDA, *s.get(),
-        *u.get(), LDU, *vt.get(), LDVt, *scratchpad.get(), scratch_size);
+    // MKL is finicky about exact scratch space size so we'll need to
+    // create sycl::buffer of exact size. if we use memAlloc, this might
+    // require a sub-buffer of a sub-buffer returned by memAlloc which is
+    // currently illegal in sycl
+    int64_t scratch_size =
+        ::oneapi::mkl::lapack::gesvd_scratchpad_size<compute_t<T>>(
+            getQueue(), ::oneapi::mkl::jobsvd::vectors,
+            ::oneapi::mkl::jobsvd::vectors, M, N, LDA, LDU, LDVt);
+
+    sycl::buffer<compute_t<T>> scratchpad(scratch_size);
```

Comment:
can you make using statements in cpp files? i don't like namespaces sprinkled everywhere.

Reply:
keeping ::oneapi::mkl since those don't make sense on their own